ENGDOCS-744 Compose Install Fixes (#14892)
docker/docker.github.io: compose/install/compose-plugin.md

---
description: How to install Docker Compose CLI plugin
keywords: compose, orchestration, install, installation, docker, documentation
toc_max: 3

title: Install Docker Compose CLI plugin
---

In this page you can find instructions on how to install Compose Plugin for Docker CLI in: 
* Linux
* Windows Server

## Compose prerequisites

* Docker Compose requires Docker Engine.
* Docker Compose plugin requires Docker CLI.

## Installing Compose on Linux systems

> In this section you can find various routes for installing Compose in Linux.

Author: Changing routes to methods because the next header includes "methods".

    ```console
    $ sudo apt-get update
    $ sudo apt-get install docker-compose-plugin
    ```
    
    Alternatively, to install a specific version of Compose CLI plugin:
      
    a. List the versions available in your repo:

      ```console
      $ apt-cache madison docker-compose-plugin

        docker-compose-plugin | 2.3.3~ubuntu-focal | https://download.docker.com/linux/ubuntu focal/stable arm64 Packages
      ```

    b. From the list obtained use the version string you can in the second column to specify the version you wish to install. 
      
    c. Install the selected version:

      ```console
      $ sudo apt-get install docker-compose-plugin=<VERSION_STRING>
      ```
    where `<VERSION_STRING>` is, for example,`2.3.3~ubuntu-focal`.

3.  Verify that Docker Compose is installed correctly by checking the version.

    ```console
    $ docker compose version
    Docker Compose version v2.3.3
    ```

### Install the plugin manually

> **Note**
>
> These instructions assume you already have Docker Engine and Docker CLI installed and now want to install the Compose plugin. 
>

> > Note as well this option requires you to manage upgrades manually. Whenever possible we recommend any of the other installation methods listed. For other Linux installation methods see [this summary](#installation-methods).

Reviewer: ```suggestion
> Installing the Compose plugin manually also requires you to manage upgrades manually. Therefore, we recommend that you use the other installation methods documented in the [Installation methods](#installation-methods) section.
```

>To run Compose as a non-root user, see [Manage Docker as a non-root user](../../engine/install/linux-postinstall.md).

1.  To download and install the Compose CLI plugin, run:

    ```console
    $ DOCKER_CONFIG=${DOCKER_CONFIG:-$HOME/.docker}
    $ mkdir -p $DOCKER_CONFIG/cli-plugins
    $ curl -SL https://github.com/docker/compose/releases/download/{{site.compose_version}}/docker-compose-linux-x86_64 -o $DOCKER_CONFIG/cli-plugins/docker-compose
    ```

    This command downloads the latest release of Docker Compose (from the Compose releases repository) and installs Compose for _the active user_ under `$HOME` directory. 
    
    > To install:

>     >* Docker Compose for _all users_ on your system, replace `~/.docker/ cli-plugins` with `/usr/local/lib/docker/cli-plugins`.

Reviewer 2: Found a minor typo in the first directory, removing one space.

```suggestion
    >* Docker Compose for _all users_ on your system, replace `~/.docker/cli-plugins` with `/usr/local/lib/docker/cli-plugins`.
```

Author: Thanks, for your comments Stefan. I have already negotiated each of Usha's suggestions offline. I have been using the "Resolve conversation" button bellow each comment to mark changes I have already addressed in my local branch before committing. So all comments are still present on the page but just collapsed.
I will add your improvements as well and then address conflicts.

Reviewer 2: I also can't see this change. Please take another look.

    >* A different version of Compose, substitute `{{site.compose_version}}` with the version of Compose you want to use.

2. Apply executable permissions to the binary:

     ```console
    $ chmod +x $DOCKER_CONFIG/cli-plugins/docker-compose
    ```
    or, if you chose to install Compose for all users:

    ```console
    $ sudo chmod +x /usr/local/lib/docker/cli-plugins/docker-compose
    ```

3. Test the installation.

    ```console
    $ docker compose version
    Docker Compose version {{site.compose_version}}
    ```

> **Note**
>

> >__Compose standalone__: If you need to use Compose without installing the Docker CLI, the instructions for the standalone scenario are similar. 

Reviewer: Should this be a new heading as we are providing instructions on how to install Compose as a standalone?

> Note the target folder for the binary's installation is different as well as the compose syntax used with the plugin (_space compose_) or the standalone version (_dash compose_).

1. To download and install Compose standalone, run:
  ```console
  $ curl -SL https://github.com/docker/compose/releases/download/{{site.compose_version}}/docker-compose-linux-x86_64 -o /usr/local/bin/docker-compose
  ```
2. Apply executable permissions to the standalone binary in the target path for the installation.
3. Test and execute compose commands using `docker-compose`.

> **Note**
>
> If the command `docker-compose` fails after installation, check your path.
> You can also create a symbolic link to `/usr/bin` or any other directory in your path. 
> For example:
> ```console
> $ sudo ln -s /usr/local/bin/docker-compose /usr/bin/docker-compose
> ```

## Install Compose on Windows Server

Follow these instructions if you are running the Docker daemon and client directly
on Microsoft Windows Server and want to install Docker Compose.

> 1.  Run a PowerShell as an administrator. 

Reviewer: ```suggestion
1.  Run PowerShell as an administrator. 
```

> When asked if you want to allow this app to make changes to your device, click **Yes** in order to proceed with the installation.

Reviewer: ```suggestion
When asked if you want to allow this app to make changes to your device, click **Yes** to proceed with the installation.
```
